### ISSUE TITLE
[BUG] Null on server start and displays don't load

### ISSUE THREAD

**Issue by PotatoIQ** [snowgears/shopbugs#240]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Type of bug

Error in console

### Environment

```markdown
- Server Version: paper 1.16.5
- Shop Version: 1.8.2.9
```

### Server startup log

https://haste.olziedev.com/ihafegoyam.sql

### Shop config files

https://haste.olziedev.com/ijagepezef.yaml

### Bug description

I got null after restarting the server, and shop displays didn't load, then I done /shop reload, and /shop item refresh, and everything worked fine without any errors.

### Steps to reproduce

1- restart your server
2- many players join in the same time
3- null pointer appears in console
4- shop displays don't show

### Expected behaviour

shops display correctly

### Actual behaviour

shop displays don't appear

**Comment by untuned** [comment]
Relevant errors:
```
[16:00:21] [User Authenticator #1/INFO]: UUID of player ForteanSlip is 6562ffa4-546a-4dbf-9da8-9690b2cd1248
[16:00:21] [User Authenticator #2/INFO]: UUID of player SoroFoxDragon is 1002d9f9-f4c7-4ef6-9b32-2090c2a8d380
[16:00:21] [User Authenticator #5/INFO]: UUID of player TatoIQ is 30b7148d-361d-44eb-97e3-5bcb0b0b9bec
[16:00:22] [User Authenticator #3/INFO]: UUID of player *Cian8246 is 00000000-0000-0000-0009-01f0d66211d2
[16:00:22] [User Authenticator #4/INFO]: UUID of player HQdoggie is 2ba66e96-9746-4175-9822-91173610513d
[16:00:22] [Server thread/INFO]: [floodgate] Floodgate player logged in as *Cian8246 joined (UUID: 00000000-0000-0000-0009-01f0d66211d2)
[16:00:22] [Server thread/WARN]: [Shop] Task #3746 for Shop v1.8.2.9 generated an exception
java.util.ConcurrentModificationException: null
	at java.util.HashMap$HashIterator.nextNode(HashMap.java:1493) ~[?:?]
	at java.util.HashMap$ValueIterator.next(HashMap.java:1521) ~[?:?]
	at com.snowgears.shop.handler.ShopHandler.refreshShopDisplays(ShopHandler.java:346) ~[?:?]
	at com.snowgears.shop.listener.ShopListener$1.run(ShopListener.java:332) ~[?:?]
	at org.bukkit.craftbukkit.v1_16_R3.scheduler.CraftTask.run(CraftTask.java:100) ~[patched_1.16.5.jar:git-Paper-778]
	at org.bukkit.craftbukkit.v1_16_R3.scheduler.CraftScheduler.mainThreadHeartbeat(CraftScheduler.java:468) ~[patched_1.16.5.jar:git-Paper-778]
	at net.minecraft.server.v1_16_R3.MinecraftServer.b(MinecraftServer.java:1427) ~[patched_1.16.5.jar:git-Paper-778]
	at net.minecraft.server.v1_16_R3.DedicatedServer.b(DedicatedServer.java:436) ~[patched_1.16.5.jar:git-Paper-778]
	at net.minecraft.server.v1_16_R3.MinecraftServer.a(MinecraftServer.java:1342) ~[patched_1.16.5.jar:git-Paper-778]
	at net.minecraft.server.v1_16_R3.MinecraftServer.w(MinecraftServer.java:1130) ~[patched_1.16.5.jar:git-Paper-778]
	at net.minecraft.server.v1_16_R3.MinecraftServer.lambda$a$0(MinecraftServer.java:291) ~[patched_1.16.5.jar:git-Paper-778]
	at java.lang.Thread.run(Thread.java:829) [?:?]
[16:00:22] [Server thread/WARN]: [Shop] Task #3760 for Shop v1.8.2.9 generated an exception
java.util.ConcurrentModificationException: null
	at java.util.HashMap$HashIterator.nextNode(HashMap.java:1493) ~[?:?]
	at java.util.HashMap$ValueIterator.next(HashMap.java:1521) ~[?:?]
	at com.snowgears.shop.handler.ShopHandler.refreshShopDisplays(ShopHandler.java:346) ~[?:?]
	at com.snowgears.shop.listener.ShopListener$1.run(ShopListener.java:332) ~[?:?]
	at org.bukkit.craftbukkit.v1_16_R3.scheduler.CraftTask.run(CraftTask.java:100) ~[patched_1.16.5.jar:git-Paper-778]
	at org.bukkit.craftbukkit.v1_16_R3.scheduler.CraftScheduler.mainThreadHeartbeat(CraftScheduler.java:468) ~[patched_1.16.5.jar:git-Paper-778]
	at net.minecraft.server.v1_16_R3.MinecraftServer.b(MinecraftServer.java:1427) ~[patched_1.16.5.jar:git-Paper-778]
	at net.minecraft.server.v1_16_R3.DedicatedServer.b(DedicatedServer.java:436) ~[patched_1.16.5.jar:git-Paper-778]
	at net.minecraft.server.v1_16_R3.MinecraftServer.a(MinecraftServer.java:1342) ~[patched_1.16.5.jar:git-Paper-778]
	at net.minecraft.server.v1_16_R3.MinecraftServer.w(MinecraftServer.java:1130) ~[patched_1.16.5.jar:git-Paper-778]
	at net.minecraft.server.v1_16_R3.MinecraftServer.lambda$a$0(MinecraftServer.java:291) ~[patched_1.16.5.jar:git-Paper-778]
	at java.lang.Thread.run(Thread.java:829) [?:?]
[16:00:22] [Server thread/WARN]: [Shop] Task #3797 for Shop v1.8.2.9 generated an exception
java.util.ConcurrentModificationException: null
	at java.util.HashMap$HashIterator.nextNode(HashMap.java:1493) ~[?:?]
	at java.util.HashMap$ValueIterator.next(HashMap.java:1521) ~[?:?]
	at com.snowgears.shop.handler.ShopHandler.refreshShopDisplays(ShopHandler.java:346) ~[?:?]
	at com.snowgears.shop.listener.ShopListener$1.run(ShopListener.java:332) ~[?:?]
	at org.bukkit.craftbukkit.v1_16_R3.scheduler.CraftTask.run(CraftTask.java:100) ~[patched_1.16.5.jar:git-Paper-778]
	at org.bukkit.craftbukkit.v1_16_R3.scheduler.CraftScheduler.mainThreadHeartbeat(CraftScheduler.java:468) ~[patched_1.16.5.jar:git-Paper-778]
	at net.minecraft.server.v1_16_R3.MinecraftServer.b(MinecraftServer.java:1427) ~[patched_1.16.5.jar:git-Paper-778]
	at net.minecraft.server.v1_16_R3.DedicatedServer.b(DedicatedServer.java:436) ~[patched_1.16.5.jar:git-Paper-778]
	at net.minecraft.server.v1_16_R3.MinecraftServer.a(MinecraftServer.java:1342) ~[patched_1.16.5.jar:git-Paper-778]
	at net.minecraft.server.v1_16_R3.MinecraftServer.w(MinecraftServer.java:1130) ~[patched_1.16.5.jar:git-Paper-778]
	at net.minecraft.server.v1_16_R3.MinecraftServer.lambda$a$0(MinecraftServer.java:291) ~[patched_1.16.5.jar:git-Paper-778]
	at java.lang.Thread.run(Thread.java:829) [?:?]
[16:00:22] [Server thread/WARN]: [Shop] Task #3802 for Shop v1.8.2.9 generated an exception
java.util.ConcurrentModificationException: null
	at java.util.HashMap$HashIterator.nextNode(HashMap.java:1493) ~[?:?]
	at java.util.HashMap$ValueIterator.next(HashMap.java:1521) ~[?:?]
	at com.snowgears.shop.handler.ShopHandler.refreshShopDisplays(ShopHandler.java:346) ~[?:?]
	at com.snowgears.shop.listener.ShopListener$1.run(ShopListener.java:332) ~[?:?]
	at org.bukkit.craftbukkit.v1_16_R3.scheduler.CraftTask.run(CraftTask.java:100) ~[patched_1.16.5.jar:git-Paper-778]
	at org.bukkit.craftbukkit.v1_16_R3.scheduler.CraftScheduler.mainThreadHeartbeat(CraftScheduler.java:468) ~[patched_1.16.5.jar:git-Paper-778]
	at net.minecraft.server.v1_16_R3.MinecraftServer.b(MinecraftServer.java:1427) ~[patched_1.16.5.jar:git-Paper-778]
	at net.minecraft.server.v1_16_R3.DedicatedServer.b(DedicatedServer.java:436) ~[patched_1.16.5.jar:git-Paper-778]
	at net.minecraft.server.v1_16_R3.MinecraftServer.a(MinecraftServer.java:1342) ~[patched_1.16.5.jar:git-Paper-778]
	at net.minecraft.server.v1_16_R3.MinecraftServer.w(MinecraftServer.java:1130) ~[patched_1.16.5.jar:git-Paper-778]
	at net.minecraft.server.v1_16_R3.MinecraftServer.lambda$a$0(MinecraftServer.java:291) ~[patched_1.16.5.jar:git-Paper-778]
	at java.lang.Thread.run(Thread.java:829) [?:?]
```

**Comment by snowgears** [comment]
Believe to have fixed in 1.8.3 (in progress)

Please reopen if this happens in the future
